### PR TITLE
Add double variable for OPC UA Protocol

### PR DESCRIPTION
**Pull request by rtorresmatheus**
Changes:

- Added flow to accept double variable;
- Added option to read double variable from OPC UA Protocol.

### CODE REVIEW

**Review by AntonioJanael — APPROVED** [review]

File: src/iface-opc-ua.c [hunk 1]
```diff
@@ -134,94 +136,145 @@ static UA_StatusCode setup_read_data(UA_Variant *value,
...
 }
 
-static UA_StatusCode get_type_uint(union opc_ua_types *tmp,
-				   void * data,
+static void get_type_uint(union opc_ua_types *tmp,
+				   UA_Variant *data,
 				   int value_type_size)
 {
-	UA_StatusCode rc = UA_STATUSCODE_GOOD;
+	if (value_type_size == 8) {
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c]: I would recommend replacing the number 8 with something like a name with more details: size_val_8.

File: src/iface-opc-ua.c [hunk 2]
```diff
@@ -240,21 +293,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
...
 		l_error("Failed to read from OPC UA: %#010x",
 			rc);
 		l_io_destroy(opc_ua_io);
 		opc_ua_io = NULL;
 		rental = -EINTR;
-	} else if (rc & UA_STATUSCODE_UNCERTAIN) {
+	} else if (((rc & UA_STATUSCODE_BAD) ||
+		   (rc & UA_STATUSCODE_UNCERTAIN))) {
 		l_error("Failed to read from OPC UA for Data_%d: %#010x",
 			data_item->sensor_id, rc);
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c]: I would recommend replacing the 'rc' and 'fd' variables with names with more details

File: src/properties.c [hunk 3]
```diff
@@ -166,11 +166,10 @@ static int set_data_properties(struct knot_thing *thing,
 	int element_size_aux = KNOT_ERR_INVALID;
 	struct knot_data_item data_item_aux;
 
-	storage_read_key_int(fd, group_id, DATA_NAME_SPACE_INDEX,
+	rc = storage_read_key_int(fd, group_id, DATA_NAME_SPACE_INDEX,
```
Comment (luisfelipemisil) [inline src/properties.c]: I would recommend replacing the 'fd' variable with a name with more details

File: src/properties.c [hunk 4]
```diff
@@ -166,11 +166,10 @@ static int set_data_properties(struct knot_thing *thing,
...
 		&namespace_aux);
-	if (namespace_aux == -EINVAL)
-		return -EINVAL;
-	data_item_aux.namespace = namespace_aux;
+	if (rc != -EINVAL)
+		data_item_aux.namespace = namespace_aux;
 
 	identifier_type_aux = storage_read_key_string(fd, group_id,
```
Comment (luisfelipemisil) [inline src/properties.c]: I would recommend replacing the 'fd' variable with a name with more details

File: src/properties.c [hunk 5]
```diff
@@ -291,6 +287,12 @@ static int get_upper_limit(int fd, char *group_id, int value_type,
 		break;
 	case KNOT_VALUE_TYPE_RAW:
 		/* Storage doesn't give support to raw */
+	case KNOT_VALUE_TYPE_DOUBLE:
+		rc = storage_read_key_double(fd, group_id,
```
Comment (luisfelipemisil) [inline src/properties.c:291]: I would recommend replacing the 'rc variable with a name with more details

**Review by luisfelipemisil — APPROVED** [review]

**Pull request merged**